### PR TITLE
fix: Added dropdown to select whether to send credentials in header or body in client credentials flow

### LINKED ISSUE #14218: [Bug]-[4400]:Sending credentials as body or header option is not provided in Authenticated API
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

In OAuth2.0 authentication there is no option to send the authentication parameters as header or body
Screenshot 2022-06-01 at 12 36 18 PM

image
Adding postman screenshot for reference
Screenshot 2022-06-01 at 12 28 40 PM

### Steps To Reproduce

1. Create a new Authenticated API in Datasource page.

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by nidhi-nair**
## Description

We were missing the option to select between body and header while sending credentials in the client credentials flow. This fix introduces the dropdown and maintains the default of sending as body for existing APIs.

Fixes #14218 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manually tested by hitting a mock endpoint with both settings
- JUnit tests for both settings
- Cypress test to check for correct form rendering and default option to send as body selected

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Parthvi12 — CHANGES_REQUESTED** [review]

File: app/client/cypress.json [hunk 1]
```diff
@@ -22,5 +22,13 @@
 	"retries": {
 		"runMode": 1,
 		"openMode": 0
-	}
+	},
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/client/cypress.json:25]: Oops :|

[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/client/cypress.json]: Done :D

**Review by Parthvi12 — DISMISSED** [review]

**Comment by Parthvi12** [comment]
Cypress tests look good to me.

**Review by somnathdasadhikari — APPROVED** [review]

**Review by sumitsum — APPROVED** [review]

**Comment by nidhi-nair** [comment]
Merging in because tests have passed but ui-test-result won't update.

**Pull request merged**